larq_compute_engine/core/bgemm_kernels_arm32.h

```diff
@@ -341,6 +350,34 @@ void BinaryKernelNeonOutOfOrder4x4(BinaryKernelParams<4, 4>& params) {
       "vadd.f32 q10, q10, q15\n"
       "vadd.f32 q11, q11, q15\n"
 
+IF_INT8_OUTPUT(
+      // Convert to signed integer, rounding to nearest.
+      //     Arm32 NEON has an instruction (vcvtr) to convert from float->int
+      // with rounding-to-nearest behaviour, but for some reason when used here
+      // truncation (rounding-to-zero) behaviour was observed instead. It's
+      // unclear what the issue is - perhaps QEMU doesn't correctly emulate the
+      // rounding behaviour?
+      //     The work-around is to first convert from float to fixed-point (24
+      // integer bits, 8 fractional bits)...
```

Comment:
Another workaround would be to add `+0.5` and then either floor or truncate.

Reply:
Yeah I thought about adding 0.5 but I think to get that to work properly it has to be +0.5 for a positive value and -0.5 for a negative value, which means it isn't possible to blindly add 0.5 to the output transform bias.

And you're right that there's a FPSCR register with a bit that you can set for the rounding mode, but according to the docs I found Neon ignores this value (at least on Arm32).

Reply:
Ah you're right, I missed that you wrote "rounding-to-zero" instead of "rounding to negative infinity" (for which +0.5 would work). The current workaround seems totally fine to me :+1: , thanks for figuring that out.

Reply:
Yeah this actually doesn't use any additional instructions compared to having `vcvtr` convert-with-rounding anyway, because the step where we shift-and-narrow-with-rounding-and-saturation the fixed-point values does the narrowing to int16 and so replaces what would have to otherwise be a block of `sqxtn` instructions (as used in the Aarch64 kernels).